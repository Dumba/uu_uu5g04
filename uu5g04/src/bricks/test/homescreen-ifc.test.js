/**
 * Copyright (C) 2019 Unicorn a.s.
 *
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 *
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

// make userAgent settable
if (
  !window.navigator.hasOwnProperty("userAgent") ||
  !Object.getOwnPropertyDescriptor(window.navigator, "userAgent").set
) {
  let userAgent = window.navigator.userAgent;
  Object.defineProperty(window.navigator, "userAgent", {
    get() {
      return userAgent;
    },
    set(value) {
      userAgent = value;
    },
  });
}

describe(`UU5.Bricks.HomeScreen interface testing`, () => {
  it("activate(setStateCallBack)", () => {
    // change userAgent to Safari on iOS 10 to pass homescreen detection
    // and use fake timers because activation is async operation
    jest.useFakeTimers();
    let origUserAgent = window.navigator.userAgent;
    window.navigator.userAgent =
      "Mozilla/5.0 (iPhone; CPU iPhone OS 10_0_1 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) Version/10.0 Mobile/14A403 Safari/602.1";
    try {
      const wrapper = shallow(
        <UU5.Bricks.HomeScreen
          screenSize={"xs"}
          id={"uuID"}
          startDelay={1000}
          message="Something"
          lifespan={30000}
          displayPace={0}
        />
      );
      expect(wrapper).toMatchSnapshot();
      const mockFunc = jest.fn();
      const returnValue = wrapper.instance().activate(mockFunc);
      jest.runAllTimers();
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(mockFunc).toBeCalled();
      expect(mockFunc).toHaveBeenCalledTimes(1);
      expect(returnValue === wrapper.instance()).toBe(true);
    } finally {
      window.navigator.userAgent = origUserAgent;
    }
  });
});
