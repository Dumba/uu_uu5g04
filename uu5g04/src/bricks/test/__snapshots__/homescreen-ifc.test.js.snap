// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.HomeScreen interface testing activate(setStateCallBack) 1`] = `
<Div
  className="ath-viewport"
  content={null}
  controlled={true}
  disabled={false}
  dynamic={false}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel="bigBoxCollection"
  noIndex={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
>
  <div
    className="ath-container ath-iOS ath-iOS10 ath-phone uu5-bricks-home-screen-hidden"
  >
    <div
      key="1"
      onClick={[Function]}
    >
      <Icon
        authenticate={false}
        className={null}
        clickable={true}
        controlled={true}
        disabled={false}
        hidden={false}
        icon="mdi-close"
        id={null}
        mainAttrs={null}
        name={null}
        nestingLevel={null}
        noIndex={false}
        pureRender={false}
        ref_={null}
        selected={false}
        src=""
        style={null}
        tooltip={null}
      />
    </div>
    <P
      className={null}
      content="Something"
      controlled={true}
      disabled={false}
      dynamic={false}
      hidden={false}
      id={null}
      ignoreInnerHTML={false}
      key="2"
      mainAttrs={null}
      name={null}
      nestingLevel={null}
      noIndex={false}
      pureRender={false}
      ref_={null}
      selected={false}
      style={null}
      tooltip={null}
    />
  </div>
</Div>
`;

exports[`UU5.Bricks.HomeScreen interface testing activate(setStateCallBack) 2`] = `
<Div
  className="ath-viewport"
  content={null}
  controlled={true}
  disabled={false}
  dynamic={false}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel="bigBoxCollection"
  noIndex={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
>
  <div
    className="ath-container ath-iOS ath-iOS10 ath-phone uu5-bricks-home-screen-hidden"
  >
    <div
      key="1"
      onClick={[Function]}
    >
      <Icon
        authenticate={false}
        className={null}
        clickable={true}
        controlled={true}
        disabled={false}
        hidden={false}
        icon="mdi-close"
        id={null}
        mainAttrs={null}
        name={null}
        nestingLevel={null}
        noIndex={false}
        pureRender={false}
        ref_={null}
        selected={false}
        src=""
        style={null}
        tooltip={null}
      />
    </div>
    <P
      className={null}
      content="Something"
      controlled={true}
      disabled={false}
      dynamic={false}
      hidden={false}
      id={null}
      ignoreInnerHTML={false}
      key="2"
      mainAttrs={null}
      name={null}
      nestingLevel={null}
      noIndex={false}
      pureRender={false}
      ref_={null}
      selected={false}
      style={null}
      tooltip={null}
    />
  </div>
</Div>
`;
